refactor: update `stats/base/dmax` native addon from C++ to C

aayush0325 opened the pull request:
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactors the native addon of `stats/base/dmax` to follow latest conventions.
-   I browsed through some old PRs ( following #1807 here ) to get an idea of what should be done here. 
-   **So far, the only change made here is migration from C++ to C addons**
-   However for refrence say #1807 has some other changes related to tests, benchmarks and documentation as well. Please let me know if we need to do the same here

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aayush0325 committed:
refactor: update stats/base/dmax to follow latest project conventions

kgryte commented on lib/node_modules/@stdlib/stats/base/dmax/manifest.json:
```diff
@@ -24,6 +26,28 @@
+      ]
+    },
+    {
+      "task": "examples",
```
You also need a configuration for benchmarks.

kgryte commented on lib/node_modules/@stdlib/stats/base/dmax/src/addon.c:
```diff
@@ -0,0 +1,46 @@
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
```
Use `STDLIB_NAPI_CREATE_DOUBLE`.

kgryte replied:
You should be able to search `blas/ext/base/*` for an example usage.

kgryte reviewed (changes requested):
This is looking good. Just two minor comments.

kgryte commented:
/stdlib merge

kgryte commented:
Let's hold off on additional changes for future PRs.

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into refactor-dmax

aayush0325 commented:
Noted, pushing the requested changes soon!

aayush0325 committed:
chore: changes from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 commented:
is this good to go @kgryte?

kgryte reviewed (approved):
LGTM. Thanks, @aayush0325!
